kv/dist_sender.go

```diff
@@ -429,6 +431,13 @@ func (ds *DistSender) Send(call *client.Call) {
 	for {
 		reply := call.Reply
 		err := util.RetryWithBackoff(retryOpts, func() (util.RetryStatus, error) {
+			select {
+			case <-ds.closer:
+				call.Reply.Header().SetGoError(util.Errorf("sender is closed"))
```

Comment:
I was assuming that this is a test-only issue - in production we'd either go crash-only or have a more elaborate shutdown process with a lame-duck period. I'll return a TransactionRetryError instead of a util.Errorf.